src/main/groovy/com/ofg/pipeline/core/PipelineBuilder.groovy

```diff
@@ -28,10 +30,13 @@ class PipelineBuilder<P extends Project> {
     private class StageContext {
         private String stageName
 
-        //TODO use the P type instead of ? extends Project in a way that does not trigger type warnings in IDEA
-        //(probably impossible without Groovy 2.4)
-        //TODO in the meantime, maybe inspect he generic type in runtime and fail fast?
-        void job(JobDefinition<? extends Job, ? extends Project> jobDefinition) {
+        void job(Optional<JobDefinition<? extends Job, P>> optionalJobDefinition) {
```

Comment:
I think those TODO's are still relevant :) Especially if you're on jenkins 2.0 which (hopefully) has groovy 2.4 :)